src/mininet_test/test_monitor.py

```diff
@@ -62,7 +62,9 @@ def stop(self):
             process = self.running[fd].process
             process.poll()
             if process.returncode:
-                assert 0
+                raise RuntimeError(
+                    "Process exited with error: returncode={} stderr={}"
+                    .format(process.returncode, process.stderr.read()))
```

Comment:
Maybe we should just pass the `self.running[fd]` to the runtime error. It will provide more information about the process:
```
raise RuntimeError("Process exited with error: {}".format(self.running[fd]))
```

Otherwise it looks good to me.